python/sglang/srt/mem_cache/radix_cache.py

```diff
@@ -429,11 +449,8 @@ def _print_helper(self, node: TreeNode, indent: int):
                 ), f"{key=}, {self.get_child_key_fn(child.key)=}"
 
     def _delete_leaf(self, node):
-        for k, v in node.parent.children.items():
-            if v == node:
-                break
-        del node.parent.children[k]
-        self.evictable_size_ -= len(node.key)
+        if node.child_key in node.parent.children:
```

Comment:
this if is not needed, instead it should be an assert, after checking `node.value is None` properly.